src/main/java/org/sonar/plugins/stash/InputFileCacheSensor.java

```diff
@@ -1,47 +1,38 @@
 package org.sonar.plugins.stash;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.sonar.api.BatchComponent;
 import org.sonar.api.batch.Sensor;
 import org.sonar.api.batch.SensorContext;
-import org.sonar.api.batch.fs.FilePredicate;
 import org.sonar.api.batch.fs.FileSystem;
 import org.sonar.api.batch.fs.InputFile;
 import org.sonar.api.resources.Project;
 
-public class InputFileCacheSensor implements Sensor {
-
-  private final StashPluginConfiguration stashPluginConfiguration;
-  private final FileSystem fileSystem;
-  private final InputFileCache inputFileCache;
-
-  public InputFileCacheSensor(StashPluginConfiguration stashPluginConfiguration, FileSystem fileSystem,
-      InputFileCache inputFileCache) {
-    this.stashPluginConfiguration = stashPluginConfiguration;
-    this.fileSystem = fileSystem;
-    this.inputFileCache = inputFileCache;
-  }
-
-  @Override
-  public void analyse(Project module, SensorContext context) {
-    for (InputFile inputFile : fileSystem.inputFiles(getAllPredicateFiles())) {
-      inputFileCache.putInputFile(context.getResource(inputFile).getEffectiveKey(), inputFile);
+/*
+ * FIXME this should not be necessary, the new plugin API gives us direct access to the InputFile of an issue
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1134)